src/lib/menu/menu-trigger.ts

```diff
@@ -459,4 +455,32 @@ export class MatMenuTrigger implements AfterContentInit, OnDestroy {
     }
   }
 
+  /** Handles the cases where the user hovers over the trigger. */
+  private _handleHover() {
+    // Subscribe to changes in the hovered item in order to toggle the panel.
+    if (!this.triggersSubmenu()) {
+      return;
+    }
+
+    this._hoverSubscription = this._parentMenu._hovered()
+      // Since we might have multiple competing triggers for the same menu (e.g. a sub-menu
+      // with different data and triggers), we have to delay it by a tick to ensure that
+      // it won't be closed immediately after it is opened.
+      .pipe(filter(active => active === this._menuItemInstance), delay(0))
```

Comment:
Will this delay trigger multiple change detections if you quickly hover over multiple menu items?

Reply:
We're triggering change detections anyway, because we have a `mouseenter` listener on each menu item.